### ISSUE TITLE
continuously decreasing callback_count ratio

### ISSUE THREAD

**Issue by Johnson-Wang** [syr-cn/ReMemR1#3]
Hi authors, 
    Congratulations on being accepted by ICLR2026 and thanks for sharing this great work. I was trying to reproduce your work but I encountered several issues. Would you please take a look and respond at your convenience?

First I found some inconsistencies in the code, and I corrected them according to my understanding of the paper. 
- https://github.com/syr-cn/ReMemR1/blob/50a64211574f917713499a3a310894c3109ab541/recurrent/impls/memory_revisit.py#L35 was modified to
  ```
      @property
      def max_raw_input_length(self):
          # Another self.max_memorization_length is for memory recall
          return self.max_prompt_length + self.chunk_size + self.max_memorization_length + self.max_memorization_length
  ```
- https://github.com/syr-cn/ReMemR1/blob/50a64211574f917713499a3a310894c3109ab541/recurrent/impls/memory_revisit.py#L170 was changed to 
  ```
          self.max_input_length = self.config.max_raw_input_length + self.token_message_template.length  # self.config.max_raw_input_length should not be counted twice as only memory blocks are recalled not full chunk
  ```

Then I was ready to run the training according to `scripts/1_run_train_ReMemR1_3B.sh`. I also used a 1.5B model to accelerate the training without changing any other hyper-parameters. However, what I observed was a continuously decreasing `callback_count` ratio as below

<img width="692" height="326" alt="Image" src="https://github.com/user-attachments/assets/8819bd83-f0be-4374-9a1d-b12181b009d5" />

I wonder if that is normal or I am missing some important details in my implementation.

Thanks

**Comment by Johnson-Wang** [comment]
Thanks for your response and the shared wandb log. The following clip is from your shared wandb log, and you can clearly identify that the callback count ratio of a 3B model keeps decreasing, which means that the trained 3B model actually degrades to a pure MemAgent.  I wonder if this statement is true from the shared training log.

<img width="733" height="326" alt="Image" src="https://github.com/user-attachments/assets/715518ee-866e-4827-a35a-cf47268a7181" />

**Comment by syr-cn** [comment]
Thank you for pointing this out!

Sorry I accidentally pasted the screenshots for `update_count` instead! X)

Both experiments are conducted with all parameters and code unchanged, except for the model scale. This phenomenon suggests that the callback ability might be more difficult for 1.5B/3B level models to learn consistently. For example, not every random seed will lead to a checkpoint with converged callback behaviors.

Maybe you could try running the experiments with a 7B model, or updating the base model to the Qwen3 series?
